shared/middleware/files_api.rb

```diff
@@ -95,6 +95,12 @@ def record_event(quota_event_type, quota_type, encrypted_channel_id)
     end
   end
 
+  set(:code_projects_domain) do |val|
+    condition do
+      (request.host == CDO.canonical_hostname('codeprojects.org')) == val
```

Comment:
This creates a Rack "condition", which is what allows routes like the ones below to be specified with `{ code_projects_domain: true }`. To evaluate that condition, it runs the code in this block (which compares the `request.host` with what we'd expect it to be if we are on that domain)